app/controllers/api/classes_controller.rb

```diff
@@ -39,7 +39,7 @@ def get_classes
     def get_new_classes
         # Classes created up to a week ago
         if @current_user
-            classes = SkillClass.where(archived: false).where("created_at >= ?", 1.week.ago)
+            classes = SkillClass.where(archived: false, deleted: false).where("created_at >= ?", 1.week.ago).where(class_params())
```

Comment:
you should be usign a rails scope https://guides.rubyonrails.org/active_record_querying.html
`SkillClass.not_deleted.where(...)`

Also, too much logic per controller x)

logic should be moved into a module or service. Then on the controller you simply do something like

```ruby
classes = SkillClassManager.new(some_params).get_new
if classes.error?
 do something
 return ...
end

classes_json = classes.data.as_json.... (only once without repeating the code)
render wtv
```